### PR TITLE
add lock info to CheckTxnStatusResponse

### PR DESCRIPTION
**Pull request by sticnarf**
For async commit recovery, we need the `min_commit_ts` of all locks. We can get secondary locks via `CheckSecondaryLocks` but we cannot get the `min_commit_ts` of the primary lock now.

We should carry the information of the whole primary lock in the response of `CheckTxnStatus`.

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -210,6 +210,7 @@ message CheckTxnStatusResponse {
     Action action = 5;
     bool use_async_commit = 6;
     repeated bytes secondaries = 7;
+    uint64 min_commit_ts = 8;
```
Comment (youjiali1995) [inline proto/kvrpcpb.proto:213]: How about returning `LockInfo` instead?

Comment (sticnarf) [reply proto/kvrpcpb.proto:213]: sounds better

**Comment by sticnarf** [comment]
Note this is an incomipatible change (but is master-to-master). Do not merge it until the corresponding changes are made and reviewed in the related projects (TiKV and unistore).

**Comment by coocood** [comment]
@sticnarf @youjiali1995 
We already have a KeyError field and that has a LockInfo, can we reuse it?

**Comment by sticnarf** [comment]
@coocood But it will be semantically strange because it's not an error for `CheckTxnStatus`...

**Comment by coocood** [comment]
LGTM

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -198,6 +198,7 @@ message CheckTxnStatusRequest {
 }
 
 message CheckTxnStatusResponse {
+    reserved 6, 7;
```
Comment (nrc) [inline proto/kvrpcpb.proto:201]: Why do we need to reserve 6 and 7, this proto has never been used in a published version, only on master, so it should be OK to make a breaking change.

Comment (sticnarf) [reply proto/kvrpcpb.proto:201]: Reserving these fields helps us pass the integration tests.
If we just change the type of the field, when we first upgrade TiKV, the old version of TiDB will receive a proto it cannot handle. And vice versa. Then, we have to force merge a TiKV PR or TiDB PR.

Comment (nrc) [reply proto/kvrpcpb.proto:201]: Can we remove them later? Seems better to force merge once, rather than live with cruft forever

Comment (sticnarf) [reply proto/kvrpcpb.proto:201]: I didn't think it's cruft... Will reserved fields have extra cost? If so, we can force merge at first.

Comment (nrc) [reply proto/kvrpcpb.proto:201]: I don't think it there is performance cost, but it is just a bit ugly to read. It is ok now, but if every proto has something like `reserved 6, 7, 10, 11, 12, 15, 16, 17;` or something

Comment (youjiali1995) [reply proto/kvrpcpb.proto:201]: Agree, it's ugly.

Comment (sticnarf) [reply proto/kvrpcpb.proto:201]: ok.

**Comment by sticnarf** [comment]
@coocood @nrc @youjiali1995 
Please also take a look at https://github.com/tikv/tikv/pull/8529 and https://github.com/ngaut/unistore/pull/418
If they look good, we can merge this PR and quickly merge  and TiDB. (to make broken time as little as possible)

File: pkg/raft_cmdpb/raft_cmdpb.pb.go [hunk 3]
```diff
@@ -71,7 +71,7 @@ func (x CmdType) String() string {
 	return proto.EnumName(CmdType_name, int32(x))
 }
 func (CmdType) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_raft_cmdpb_5dc25bc5f43956ae, []int{0}
+	return fileDescriptor_raft_cmdpb_8fa9c3cdf0171bc7, []int{0}
```
Comment (youjiali1995) [inline pkg/raft_cmdpb/raft_cmdpb.pb.go:74]: Why this file is changed?

Comment (sticnarf) [reply pkg/raft_cmdpb/raft_cmdpb.pb.go:74]: I don't know. I cleaned and regenerate, but this file is also changed.

Comment (sticnarf) [reply pkg/raft_cmdpb/raft_cmdpb.pb.go:74]: I use protoc 3.8.0 following the README. Maybe the last commit didn't use the expected version?

**Review by youjiali1995 — APPROVED** [review]

File: pkg/raft_cmdpb/raft_cmdpb.pb.go (hunk 3 above)
Comment (sticnarf) [reply pkg/raft_cmdpb/raft_cmdpb.pb.go:74]: I think it's because https://github.com/pingcap/kvproto/pull/644 didn't generate the go proto.

**Review by MyonKeminta — APPROVED** [review]

File: proto/kvrpcpb.proto [hunk 4]
```diff
@@ -208,8 +208,7 @@ message CheckTxnStatusResponse {
     uint64 commit_version = 4;
     // The action performed by TiKV (and why if the action is to rollback).
     Action action = 5;
-    bool use_async_commit = 6;
-    repeated bytes secondaries = 7;
+    LockInfo lock_info = 6;
```
Comment (overvenus) [inline proto/kvrpcpb.proto:211]: Reusing the same tag is a breaking change.

File: proto/kvrpcpb.proto (hunk 2 above)
Comment (overvenus) [reply proto/kvrpcpb.proto]: It's indeed a breaking change and forces BR to upgrade kvproto and TiDB https://github.com/pingcap/br/pull/481.

I suggest before we make any change to kvproto we should first make sure the feature development and test are almost complete.

Comment (nrc) [reply proto/kvrpcpb.proto]: > I suggest before we make any change to kvproto we should first make sure the feature development and test are almost complete.

It seems like we should have some conventions, certianly

File: proto/kvrpcpb.proto (hunk 4 above)
Comment (MyonKeminta) [reply proto/kvrpcpb.proto:211]: It's ok since it's only on master. But I'm afraid if other projects need to struggle to get them compiled.

**Pull request merged**